Use Google CSE for Search

alexwolfe opened the pull request:
@naomiblack Ok, this one should be ready to go. Please take a look. Thanks!

IgorMinar commented:
hi @alexwolfe, I see the following error in console when using the app with this PR: 

```
Refused to display 'https://cse.google.com/cse_v2/ads?q=template&r=m&cx=012747927672895622801%3…%2Fdocs%2Fts%2Flatest%2F&referer=http%3A%2F%2Flocalhost%3A3000%2F#master-1' in a frame because it set 'X-Frame-Options' to 'SAMEORIGIN'.
```

Can this be resolved?

IgorMinar commented on public/_includes/_scripts-minimum.jade:
```diff
@@ -18,6 +18,16 @@ script.
 
```
can you remove the swifty script? we shouldn't load it if we are not using it any more.

IgorMinar commented on public/resources/css/module/_side-nav.scss:
```diff
@@ -79,37 +79,37 @@ $sidenav-width: 240px;
+  //     color: $blue-grey-100;
+  //     font-size: 12px;
+  //   }
+  // }
```
if we no longer need this css we should delete them. otherwise it's unclear why they are just commented out.

if we do need them we should add a comment why this is being commented out temporarily and what action is being taken to uncomment this and by whom.

alexwolfe commented:
Hey @IgorMinar I addressed your comments. In terms of the X-frame origin error, that is a setting that needs to be adjust on the site (Firebase). It should be set to: 
`X-Frame-Options: ALLOW-FROM https://www.google.com` 
where that option is available.

IgorMinar commented:
@alexeagle do you know what's the impact on the site if these resources are blocked? the basic
functionality seems to be working in spite of the errors. but it is concerning that there are errors at all.

all the firebase configuration is here: https://github.com/angular/angular.io/blob/master/firebase.json, so if you believe that the change you propose above is the right one then please make the change to this json file. thanks

alexeagle commented:
@alexwolfe I think Igor's above comment is for you

IgorMinar commented:
Yes! Sorry!

On Thu, Jan 26, 2017, 7:01 AM Alex Eagle <notifications@github.com> wrote:

> @alexwolfe <https://github.com/alexwolfe> I think Igor's above comment is
> for you
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/angular/angular.io/pull/3139#issuecomment-275409581>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AANM6L_sHuLmA00UXsYnCZstJ4nwnlw1ks5rWLUwgaJpZM4Lt0E3>
> .
>

alexwolfe commented:
@IgorMinar Updated the firebase.json with the new x-frame options allowing google. Looks like the search worked either way but agree it's preferable not to have these errors.

wardbell commented:
Closing because we aren't developing the current angular.io in this repo.
